extensions/outline-document-search/src/api/OutlineApi.ts

```diff
@@ -0,0 +1,286 @@
+interface CollectionsResponse {
+  data: Collection[];
+  pagination: {
+    limit: number;
+    offset: number;
+  };
+}
+
+interface CreateDocumentResponse {
+  data: OutlineDocument;
+}
+
+/**
+ * Centralized API client for Outline
+ * Works with the Instance interface from the base extension
+ */
+export class OutlineApi {
+  private instance: Instance;
+
+  constructor(instance: Instance) {
+    this.instance = instance;
+  }
+
+  private async request<T>(method: string, body: object): Promise<T> {
+    const response = await fetch(`${this.instance.url}/api/${method}`, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${this.instance.apiKey}`,
+      },
+      body: JSON.stringify(body),
+    });
+
+    if (!response.ok) {
+      throw new Error(`Outline API error: ${response.statusText}`);
+    }
+
+    return (await response.json()) as T;
+  }
+
+  getDocumentUrl(doc: OutlineDocument): string {
+    // Outline API returns a relative URL like "/doc/title-urlId"
+    // We need to prepend the base URL
+    if (doc.url) {
+      // If url starts with http, it's already absolute
+      if (doc.url.startsWith("http")) {
+        return doc.url;
+      }
+      // Otherwise prepend base URL
+      return `${this.instance.url}${doc.url}`;
+    }
+
+    // Fallback: construct URL from id if url field is missing
+    return `${this.instance.url}/doc/${doc.id}`;
+  }
+
+  async searchDocuments(query: string): Promise<OutlineDocument[]> {
+    const response = await this.request<SearchResponse>("documents.search", {
+      query,
+      limit: 20,
+    });
+    // Extract the document from each search result
+    return response.data.map((result) => result.document);
+  }
+
+  async listDocuments(): Promise<OutlineDocument[]> {
+    const response = await this.request<{ data: OutlineDocument[] }>("documents.list", {
+      limit: 100,
+    });
+    return response.data;
+  }
+
+  async listCollections(): Promise<Collection[]> {
+    const response = await this.request<CollectionsResponse>("collections.list", {
+      limit: 100,
+    });
+    return response.data;
+  }
+
+  async createDocument(title: string, collectionId: string, text?: string): Promise<OutlineDocument> {
+    const response = await this.request<CreateDocumentResponse>("documents.create", {
+      title,
+      collectionId,
+      text: text || "",
+      publish: true,
+    });
+    return response.data;
+  }
+
+  // Starred documents
+  async getStarredDocuments(): Promise<OutlineDocument[]> {
+    const response = await this.request<{
+      data: { document: OutlineDocument }[];
+    }>("stars.list", {
+      limit: 100,
+    });
+    return response.data.map((item) => item.document);
+  }
+
+  async starDocument(id: string): Promise<void> {
+    await this.request("documents.star", { id });
+  }
+
+  async unstarDocument(id: string): Promise<void> {
+    await this.request("documents.unstar", { id });
+  }
```

Comment:
Is starring through the API backwards-compatible? I tried this against an instance on v0.86.1, where retrieving and starring documents through the extension does not work.